### PR TITLE
Let maximum transactions vary depending on block index

### PR DESCRIPTION
**Pull request by dahlia**
`IBlockPolicy<T>.MaxTransactionsPerBlock` property is replaced by `IBlockPolicy<T>.GetMaxTransactionsPerBlock(long index)` method.

### CODE REVIEW

**Comment by longfin** [comment]
It needs to rebase. @dahlia

File: Libplanet/Blockchain/Policies/BlockPolicy.cs [hunk 1]
```diff
@@ -36,7 +38,7 @@ public class BlockPolicy<T> : IBlockPolicy<T>
         /// <see cref="MinimumDifficulty"/>. 1024 by default.</param>
         /// <param name="difficultyBoundDivisor">Configures
         /// <see cref="DifficultyBoundDivisor"/>. 128 by default.</param>
-        /// <param name="maxTransactionsPerBlock">Configures <see cref="MaxTransactionsPerBlock"/>.
+        /// <param name="maxTransactionsPerBlock">Configures <see cref="_maxTransactionsPerBlock"/>.
```
Comment (limebell) [inline Libplanet/Blockchain/Policies/BlockPolicy.cs:41]: It is okay to reference private property?

Comment (dahlia) [reply Libplanet/Blockchain/Policies/BlockPolicy.cs]: Fixed them!

**Comment by dahlia** [comment]
@libplanet Rebased on the up-to-date main.  Please review this again!

File: CHANGES.md [hunk 2]
```diff
@@ -129,6 +134,9 @@ To be released.
 [#1470]: https://github.com/planetarium/libplanet/pull/1470
 [#1474]: https://github.com/planetarium/libplanet/pull/1474
 [#1475]: https://github.com/planetarium/libplanet/pull/1475
+=======
+[#1447]: https://github.com/planetarium/libplanet/pull/1446
+>>>>>>> a19776d5c (Let max transactions vary depending on block index)
```
Comment (greymistcube) [inline CHANGES.md:139]: Some leftovers. 😶

Comment (dahlia) [reply CHANGES.md]: Fixed!

**Review by greymistcube — APPROVED** [review]

**Pull request merged**